server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -206,7 +206,10 @@ public IndexService(
                 this.indexSortSupplier = () -> indexSettings.getIndexSortConfig()
                     .buildIndexSort(
                         mapperService::fieldType,
-                        (fieldType, searchLookup) -> indexFieldData.getForField(fieldType, FieldDataContext.noRuntimeFields("index sort"))
+                        (fieldType, searchLookup, fielddataOperation) -> indexFieldData.getForField(
```

Comment:
I don't think we need to pass this extra fielddataOperation parameter here?

Reply:
Fixed. Also, for my understanding, is this because we never want sort to fallback so there's no reason to build a different operation type into an index sort?

Reply:
Yes, exactly